scipy/optimize/tests/test_zeros.py

```diff
@@ -961,5 +960,89 @@ def test_maxiter_int_check_gh10236(method):
     # gh-10236 reported that the error message when `maxiter` is not an integer
     # was difficult to interpret. Check that this was resolved (by gh-10907).
     message = "'float' object cannot be interpreted as an integer"
-    with pytest.raises(TypeError, match=message):
-        method(f1, 0.0, 1.0, maxiter=72.45)
+    if method != zeros._muller:
+        with pytest.raises(TypeError, match=message):
+            method(f1, 0.0, 1.0, maxiter=72.45)
+    else:
+        with pytest.raises(TypeError, match=message):
+            method(f1, 0.0, 1.0, 2.0, maxiter=72.45)
+
+class Test_muller(TestScalarRootFinders):
+    def test__muller_real_roots(self):
+        for f in [f1, f2]:
+            x = zeros._muller(f, x0=3, x1=4, x2=5, tol=1e-6)
```

Comment:
Jake, do you have an opinion about testing the private function? I would prefer to test the `root_scalar` interface.

Reply:
Agree